src/context/TransactionsProvider/TransactionsProvider.tsx

```diff
@@ -90,29 +71,50 @@ export const TransactionsProvider = ({ children }: TransactionsProviderProps): J
           })
           .map(async chain => {
             const adapter = chainAdapterManager.byChain(chain)
-            // this looks funky, but we need a non zero length array to map over
-            const supportedAccountTypes = adapter.getSupportedAccountTypes?.() ?? [undefined]
             const chainId = adapter.getChainId()
 
             // assets are known to be defined at this point - if we don't have the fee asset we have bigger problems
             const asset = assets[chainIdToFeeAssetId(chainId)]
 
-            // TODO(0xdef1cafe) - once we have restful tx history for all coinstacks
-            // this state machine should be removed, and managed by the txHistory RTK query api
-            dispatch(txHistory.actions.setStatus('loading'))
+            // subscribe to new transactions for all supported accounts
             try {
+              // account types are only supported for utxo chains, default to undefined if no accountTypes are supported
+              const supportedAccountTypes = adapter.getSupportedAccountTypes?.() ?? [undefined]
+
               await Promise.all(
                 supportedAccountTypes.map(async accountType => {
-                  const accountParams = accountType ? utxoAccountParams(asset, accountType, 0) : {}
                   moduleLogger.info({ chainId, accountType }, 'subscribing txs')
+
+                  const accountParams = accountType ? utxoAccountParams(asset, accountType, 0) : {}
+
                   return adapter.subscribeTxs(
                     { wallet, accountType, ...accountParams },
                     msg => {
-                      const accountSpecifier = `${msg.chainId}:${msg.address}`
                       const state = store.getState()
                       const accountId = selectAccountIdByAddress(state, {
-                        accountSpecifier,
+                        accountSpecifier: `${msg.chainId}:${msg.address}`,
                       })
+
+                      const { getAccount } = portfolioApi.endpoints
+                      const { getValidatorData } = validatorDataApi.endpoints
+
+                      // refetch validator data on new txs in case TVL or APR has changed
+                      if (msg.chainId === cosmosChainId) {
```

Comment:
@elmutt FYI you'll have to update this here when osmosis stuff is merged to check for both chains